app/models/notification.rb

```diff
@@ -26,58 +26,79 @@ class Notification < ApplicationRecord
 
   paginates_per 20
 
+  API_ATTRIBUTE_MAP = {
+    repository_id: [:repository, :id],
+    repository_full_name: [:repository, :full_name],
+    repository_owner_name: [:repository, :owner, :login],
+    subject_title: [:subject, :title],
+    subject_type: [:subject, :type],
+    subject_url: [:subject, :url],
+    reason: [:reason],
+    unread: [:unread],
+    updated_at: [:updated_at],
+    last_read_at: [:last_read_at],
+    url: [:url],
+    github_id: [:id]
+  }.freeze
+
   class << self
     def download(user)
       timestamp = Time.current
 
-      fetch_notifications(user) do |notification|
+      fetch_unread_notifications(user)
+      fetch_read_notifications(user)
+
+      user.update_column(:last_synced_at, timestamp)
+    end
+
+    def attributes_from_api_response(api_response)
+      attrs = API_ATTRIBUTE_MAP.map do |attr, path|
+        [attr, api_response.to_h.dig(*path)]
+      end.to_h
+      if "RepositoryInvitation" == api_response.subject.type
+        attrs[:subject_url] = "#{api_response.repository.html_url}/invitations"
+      end
+      attrs
+    end
+
+    private
+
+    def process_unread_notifications(notifications, user)
```

Comment:
I extracted this and `process_read_notifications` with the idea of testing them separately.  I ended up not going that way with testing but left them separate.